pkg/checkpoint/checkpoint.go

```diff
@@ -0,0 +1,117 @@
+package checkpoint
+
+import (
+	"fmt"
+
+	"github.com/filecoin-project/mir/pkg/crypto"
+	"github.com/filecoin-project/mir/pkg/pb/checkpointpb"
+	"github.com/filecoin-project/mir/pkg/pb/commonpb"
+	"github.com/filecoin-project/mir/pkg/serializing"
+	t "github.com/filecoin-project/mir/pkg/types"
+)
+
+// StableCheckpoint represents a stable checkpoint.
+type StableCheckpoint checkpointpb.StableCheckpoint
```

Comment:
Is there a `NewStableCheckpoint` function that goes from the arguments of a checkpoint to an instantiation of `StableCheckpoint`? This could be really handy to perform type translations between different data structures.

Reply:
I see the point and I agree, but since the checkpoint is a rather complex data structure that is composed of different parts created at completely different times and places (that are protobufs themselves), unless I changed all that, the arguments would still need to be protobufs. I can add the `NewStableCheckpoint` function just to have it there already, but dealing with protobufs for now is almost inevitable.

Reply:
Is not critical for spacenet, I am adding my own abstraction to make them CBOR-serializable here: https://github.com/consensus-shipyard/lotus/blob/b5b81dae0c451bf26c183d17795d38c38a460a37/chain/consensus/mir/types.go#L126-L190

It adds a `serde` overhead, but I think it is worth (and low) for now.